packages/core/src/codewhispererChat/tools/chatStream.ts

```diff
@@ -22,28 +24,49 @@ export class ChatStream extends Writable {
         private readonly tabID: string,
         private readonly triggerID: string,
         private readonly toolUse: ToolUse | undefined,
+        private readonly session: ChatSession,
+        private readonly messageIdToUpdate: string | undefined,
+        // emitEvent decides to show the read toolMessage to the user.
+        private readonly emitEvent: boolean,
         private readonly validation: CommandValidation,
         private readonly changeList?: Change[],
         private readonly logger = getLogger('chatStream')
     ) {
         super()
-        this.logger.debug(`ChatStream created for tabID: ${tabID}, triggerID: ${triggerID}`)
-        this.messenger.sendInitalStream(tabID, triggerID)
+        this.logger.debug(
+            `ChatStream created for tabID: ${tabID}, triggerID: ${triggerID}, session: ${session.readFiles}, emitEvent to mynahUI: ${emitEvent}`
+        )
+        if (toolUse?.name === ToolType.FsRead && emitEvent) {
+            if (!messageIdToUpdate) {
+                // If messageIdToUpdate is undefined, we need to first create an empty message
+                // with messageId so it can be updated later
+                this.messenger.sendInitialToolMessage(tabID, triggerID, toolUse?.toolUseId)
+            }
```

Comment:
```suggestion
        if (toolUse?.name === ToolType.FsRead && emitEvent && !messageIdToUpdate) {
              // If messageIdToUpdate is undefined, we need to first create an empty message
              // with messageId so it can be updated later
              this.messenger.sendInitialToolMessage(tabID, triggerID, toolUse?.toolUseId)
```